### PR TITLE
Player controller cleanup

### PR DESCRIPTION
**Pull request by ataulien**
Refactors the player controller by removing outdated code and moving the equipment logic to its own class.

### CODE REVIEW

**Comment by frabert** [comment]
Doesn't crash on my side :)

**Comment by ataulien** [comment]
have you looked through the code? If you think it's fine, feel free to merge it!

File: src/logic/CharacterEquipment.cpp [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+bool Logic::CharacterEquipment::equipMelee(ItemHandle item)
+{
+    setItemInSlot(item, Slot::MELEE);
+
+    return true;
+}
+
+bool Logic::CharacterEquipment::equipBow(ItemHandle item)
+{
+    using Daedalus::GEngineClasses::C_Item;
```
Comment (markusobi) [inline src/logic/CharacterEquipment.cpp]: unused using, can be removed

File: src/logic/CharacterEquipment.cpp [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+
+bool Logic::CharacterEquipment::equipBelt(ItemHandle item)
+{
+    setItemInSlot(item, Slot::BELT);
+    return true;
+}
+
+bool Logic::CharacterEquipment::equipArmor(ItemHandle item)
+{
+    using Daedalus::GEngineClasses::C_Item;
```
Comment (markusobi) [inline src/logic/CharacterEquipment.cpp]: unused using, can be removed

**Comment by markusobi** [comment]
Apart from the unused using statements, you are good to go from my side.

**Pull request merged**